features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-core.tsx

```diff
@@ -0,0 +1,962 @@
+    const [ socialDisclaimerModalType, setSocialDisclaimerModalType ] = useState<LoginFlowTypes>(undefined);
+    const [ authenticators, setAuthenticators ] = useState<GenericAuthenticatorInterface[][]>(undefined);
+    const [ googleAuthenticators, setGoogleAuthenticators ] = useState<GenericAuthenticatorInterface[]>(undefined);
+    const [ gitHubAuthenticators, setGitHubAuthenticators ] = useState<GenericAuthenticatorInterface[]>(undefined);
+    const [ facebookAuthenticators, setFacebookAuthenticators ] = useState<GenericAuthenticatorInterface[]>(undefined);
+    const [ microsoftAuthenticators, setMicrosoftAuthenticators ] =
+    useState<GenericAuthenticatorInterface[]>(undefined);
+    const [ appleAuthenticators, setAppleAuthenticators ] = useState<GenericAuthenticatorInterface[]>(undefined);
+    const [ showMissingSocialAuthenticatorModal, setShowMissingSocialAuthenticatorModal ] = useState<boolean>(false);
+    const [ isAuthenticatorsFetchRequestLoading, setIsAuthenticatorsFetchRequestLoading ] = useState<boolean>(true);
+    const [
+        showDuplicateSocialAuthenticatorSelectionModal,
+        setShowDuplicateSocialAuthenticatorSelectionModal
+    ] = useState<boolean>(false);
+    const [
+        selectedSocialAuthenticator,
+        setSelectedSocialAuthenticator
+    ] = useState<GenericAuthenticatorInterface>(undefined);
+    const [
+        moderatedAuthenticationSequence,
+        setModeratedAuthenticationSequence
+    ] = useState<AuthenticationSequenceInterface>(authenticationSequence);
+    const [ selectedIDPTemplate, setSelectedIDPTemplate ] = useState<IdentityProviderTemplateInterface>(undefined);
+    const [ idpTemplateTypeToTrigger, setIDPTemplateTypeToTrigger ] = useState<string>(undefined);
+    const [ showIDPCreateWizard, setShowIDPCreateWizard ] = useState<boolean>(false);
+    const [
+        idpCreateWizardTriggerOrigin,
+        setIDPCreateWizardTriggerOrigin
+    ] = useState<"INTERNAL" | "EXTERNAL">(undefined);
+
+    const eventPublisher: EventPublisher = EventPublisher.getInstance();
+
+    /**
+     * Loads federated authenticators and local authenticators on component load.
+     */
+    useEffect(() => {
+
+        fetchAndCategorizeAuthenticators().finally();
+
+        // Reset the IDs in local storage on mount.
+        LocalStorageUtils.setValueInLocalStorage(
+            ApplicationManagementConstants.AUTHENTICATORS_LOCAL_STORAGE_KEY,
+            JSON.stringify([])
+        );
+
+    }, []);
+
+    /**
+     * If the original `authenticationSequence` changes, the moderated one should also change.
+     *
+     * @remarks This had to be added to fix the state issues came after the loading spinner
+     * conditional rendering. Components are getting unmounted when the spinner is shown causing it to
+     * re-render the component.
+     */
+    useEffect(() => {
+
+        if (authenticationSequence === undefined) {

```

Comment:
```suggestion
        if (!authenticationSequence) {

```

Reply:
Since this is a refactor, lets not change any logic.
@Achintha444 @chamathns